raster-foundry/raster-foundry
# Webpack 4 import fixes

## Overview
Fix a bunch of misc webpack errors:
* Use async webpack requires for joint.js, math.js, and plyr.js
* Split out vendor and main app bundles
* Fix favicons

### Checklist

- ~Description of PR is in an appropriate section of the [changelog](https://github.com/raster-foundry/raster-foundry/blob/develop/CHANGELOG.md) and grouped with similar changes if possible~ changelog will be edited in anchor pr
- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any new SQL strings have tests~

### Demo:
Build times on develop:
![image](https://user-images.githubusercontent.com/4392704/49240772-08bcb000-f3d4-11e8-89b5-26ec33911663.png)
Build times on webpack 4:
![image](https://user-images.githubusercontent.com/4392704/49240797-1a9e5300-f3d4-11e8-88fe-bf9f94252583.png)


## Testing Instructions
* Use a modern version of node (>8)
* Build the app using `yarn build-nginx` and verify that the bundle is now found in `app-frontend/dist` and `nginx/srv/dist`
* Verify that all favicon related files (images, manifest, etc) are present in `dist/favicons`
* Verify that there are no weird import errors for the manifest file
* re-build your nginx-api container, and start everything using `./scripts/server`
* Load `localhost:9100` and verify that bundle size is reduced, and code is split over multiple bundles.
* Turn caching off in dev tools and verify that favicons are fetched correctly
* Delete `app-frontend/dist` and verify that favicons etc are still served by the dev server
* Create a new template, and verify that the math.js bundle is fetched on demand while preserving functionality
* Verify that the lab loads correctly in the previous step, then reload the page so the lab is the first view loaded and verify that it initializes correctly.
* Verify that plyr is loaded on demand. See https://github.com/raster-foundry/raster-foundry/pull/3083 for more detailed testing instructions
* Verify that scene uploads still work (tests loam.js)
* Go to a bunch of places in the app and look for broken image imports

Closes #4200 
Closes #4204
Closes https://github.com/azavea/raster-foundry-platform/issues/527
Depends on #4325 (branched off of, rebase once that's merged)
